src/library/autodj/autodjprocessor.cpp

```diff
@@ -1216,15 +1228,20 @@ void AutoDJProcessor::calculateTransition(DeckAttributes* pFromDeck,
         introStart = toDeckPositionSeconds;
     }
 
-    double introEnd = getIntroEndSecond(pToDeck);
-    if (introEnd < introStart) {
-        // introEnd is invalid. Assume a zero length intro.
-        // The introStart is automatically placed by AnalyzerSilence, so use
-        // that as a fallback if the user has not placed introEnd.
-        introEnd = introStart;
+    double introLength = 0;
+    const double introEndSample = pToDeck->introEndPosition();
+    if (introEndSample != Cue::kNoPosition) {
+        const double introEnd = samplePositionToSeconds(introEndSample, pToDeck);
+        if (introStart < introStart) {
```

Comment:
?

Reply:
Ups, I have messed up the rebase.
I will rebase again.